src/main/scala/io/sdkman/changelogs/JavaMigrations.scala

```diff
@@ -304,4 +304,32 @@ class JavaMigrations {
   def migrate065(implicit db: MongoDatabase) = {
     Seq(Linux64, Windows, MacOSX).foreach(platform => removeVersion(candidate = "java", version = "11.e.26-openjdk", platform))
   }
+
+  @ChangeSet(order = "066", id = "065-add_eclipsej9_openjdk_8", author = "munukutla")
+  def migrate066(implicit db: MongoDatabase) = {
+    List(
+      Version("java", "8-open-j9", "https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u192-b12_openj9-0.11.0/OpenJDK8U-jdk_x64_linux_openj9_8u192b12_openj9-0.11.0.tar.gz", Linux64),
```

Comment:
We will need to follow the existing naming convention for Java candidates. Currently, this comes in the form of `major.minor.patch-distro`. As an example, this one would be `8.0.192-adopt`. We have a hard limit of 15 characters per version number, the reason being that the console column view in `sdk ls java` will truncate the version string if longer. I would suggest the `distro` part of the version string to be something concise like `adopt`, as this is currently about the party vending the software.

Reply:
So can we ignore the fact that AdoptOpenJDK also has distributions with HotSpot VM? Because `8.0.192-adopt` is available with both J9 and Hotspot.

Reply:
Yes, it would be good if we can represent this somehow, but not sure how we can pull that off with the limited characters available. Unless we find a way to abbreviate the distro to capture both. Like `8.0.192-adptj9` or something similar.

Reply:
I can foresee a number of OpenJDK distributions being released in the next couple of years. Barring any coincidences, many of them might end up having longer distribution names, like Amazon Corretto etc. Can we look to extend the 15 character limit so that it doesn't pin our heels later?

OpenJDK + Hotspot => `8.0.192-openhtspt` (it's already weird)
OpenJDK + OpenJ9 => `8.0.192-openj9` (kinda obvious)

Reply:
We would need to cross that bridge when we get there. AFAIK openjk does not have any plans in shipping anything besides hotspot vms.

Changing the length is an expensive exercise because we would need to update the list view, the vendor publishing api as well as validations in the db migrations. For me this would be premature optimisation. Besides, 15 characters should be ample for version string.

Corretto will be a single distribution and will get the suffix `amzn` or something similar. We already have _all_ the current vendors represented including graal, openjdk, zulu and oracle. I'm confident we'll manage with the scheme we have.

In future, let's please try having these discussion in Gitter prior to raising a PR.